src/NBitcoin/UInt256.cs

```diff
@@ -291,7 +291,9 @@ public override bool Equals(object obj)
         {
             if (ReferenceEquals(a, b))
                 return true;
-            if (((object)a == null) || ((object)b == null))
+            if (a is null || b is null)
```

Comment:
Should we not do the same for `Equals`?

Reply:
Yes good point - I will add the logic there as well